make a command for auto calculating the tick per inch

veronayue opened the pull request.

veronayue committed:
make a command for auto calculating the tick per inch

JohnGilb commented on Competition2018/src/main/java/competition/operator_interface/OperatorCommandMap.java:
```diff
@@ -23,8 +24,15 @@ public void setupMyCommands(
-			) {
+			ToggleGearCommand shiftGear) {
 		oi.gamepad.getifAvailable(4).whenPressed(shiftGear);
+		
```
remove unnecessary blank line

JohnGilb committed:
Merged 'master'.

JohnGilb commented on Competition2018/src/main/java/competition/subsystems/elevator/ElevatorSubsystem.java:
```diff
@@ -113,14 +115,32 @@ public double currentHeight() {
 		return ticksToInches(motor.getSelectedSensorPosition(0));
 	}
 	
+	public double currentTick() {
```
rename to `getCurrentTick()`

JohnGilb commented on Competition2018/src/main/java/competition/subsystems/elevator/ElevatorSubsystem.java:
```diff
@@ -113,14 +115,32 @@ public double currentHeight() {
+	}
+	
+	public double minHeightInInches() {
+		return ticksToInches(motor.getSelectedSensorPosition(0));
```
You need to actually get the min height in inches, which would be
` minHeightInInches.get() `

JohnGilb commented on Competition2018/src/main/java/competition/subsystems/elevator/ElevatorSubsystem.java:
```diff
@@ -113,14 +115,32 @@ public double currentHeight() {
+	}
+	
+	public double maxHeightInInches() {
+	        return ticksToInches(motor.getSelectedSensorPosition(0));
```
same with max

JohnGilb commented on Competition2018/src/main/java/competition/subsystems/elevator/ElevatorSubsystem.java:
```diff
@@ -113,14 +115,32 @@ public double currentHeight() {
+	        return ticksToInches(motor.getSelectedSensorPosition(0));
+	}
+	
+	public void setTickPerInch(double tip) {
```
`setTicksPerInch(double ticksPerInch)`

JohnGilb commented on Competition2018/src/main/java/competition/subsystems/elevator/ElevatorSubsystem.java:
```diff
@@ -113,14 +115,32 @@ public double currentHeight() {
+	}
+	
+	public void setTickPerInch(double tip) {
+	    setTickPerInch(tip);
```
this just means the method calls itself infinitely.

JohnGilb commented on Competition2018/src/main/java/competition/subsystems/elevator/commands/CalibrateCommand.java:
```diff
@@ -0,0 +1,60 @@
+import competition.subsystems.elevator.ElevatorSubsystem;
+import xbot.common.command.BaseCommand;
+
+public class CalibrateCommand extends BaseCommand {
```
name is too generic - 
`CalculateElevatorTicksPerInchCommand`

JohnGilb commented on Competition2018/src/main/java/competition/subsystems/elevator/commands/CalibrateCommand.java:
```diff
@@ -0,0 +1,60 @@
+
+    @Inject
+    public CalibrateCommand(ElevatorSubsystem elevator) {
+        // TODO Auto-generated constructor stub
```
remove any `// TODO ....`

JohnGilb commented on Competition2018/src/main/java/competition/subsystems/elevator/commands/CalibrateCommand.java:
```diff
@@ -0,0 +1,60 @@
+
+    @Override
+    public void initialize() {
+        // TODO Auto-generated method stub
```
like this one

JohnGilb commented on Competition2018/src/main/java/competition/subsystems/elevator/commands/CalibrateCommand.java:
```diff
@@ -0,0 +1,60 @@
+        }
+        
+        // Directly control the elevator with a joystick
+        elevator.setPower(0);
```
Need to get a joystick from the OI

JohnGilb commented on Competition2018/src/main/java/competition/subsystems/elevator/commands/CalibrateCommand.java:
```diff
@@ -0,0 +1,60 @@
+    @Inject
+    public CalibrateCommand(ElevatorSubsystem elevator) {
+        // TODO Auto-generated constructor stub
+        this.elevator = elevator;
```
This command probably needs to require all subsystems to make sure that the joystick it wants to use isn't in use already by other systems.

JohnGilb commented on Competition2018/src/main/java/competition/subsystems/elevator/commands/CalibrateCommand.java:
```diff
@@ -0,0 +1,60 @@
+        double ticksPerInch = ((maxTick - minTick) / (maxHeightInInches - minHeightInInches));
+        // some code here to set the ticks per inch on the subsystem
+        elevator.setTickPerInch(ticksPerInch);
+        
```
remove unnecessary blank line

veronayue committed:
changing in process

veronayue committed:
Fixed the problems:change the name, fix the oi part and require part

JohnGilb committed:
Merged 'master'.

# Conflicts:
#	Competition2018/src/main/java/competition/operator_interface/OperatorCommandMap.java
#	Competition2018/src/main/java/competition/subsystems/elevator/ElevatorSubsystem.java

JohnGilb committed:
Calibration should take command of all subsystems.

JohnGilb committed:
Have the elevator always expose its current ticks and height for the dashboard

JohnGilb committed:
Use mag encoder

JohnGilb committed:
-Use newer commonlib
-Set Elevator as a periodic data source, register in Robot
-Calibrate command is now "whileHeld)

JohnGilb committed:
Try caching gradle stuff
